lib/iris/analysis/_scipy_interpolate.py

```diff
@@ -138,20 +146,53 @@ def __call__(self, xi, method=None):
         Parameters
         ----------
         xi : ndarray of shape (..., ndim)
-            The coordinates to sample the gridded data at
+            The coordinates to sample the gridded data at.
 
         method : str
             The method of interpolation to perform. Supported are "linear" and
             "nearest".
 
         """
-        method = self.method if method is None else method
-        if method not in ["linear", "nearest"]:
-            raise ValueError("Method '%s' is not defined" % method)
+        # Note: No functionality should live in this method. It should all be
+        # decomposed into the two interfaces (compute weights + use weights).
+        weights = self.compute_interp_weights(xi, method)
+        return self.interp_using_pre_computed_weights(weights)
+
+    def compute_interp_weights(self, xi, method=None):
+        """
+        Prepare the interpolator for interpolation to the given sample points.
 
+        .. note::
+            For normal interpolation, simply call the interpolator with the
```

Comment:
It's not clear in the docstring what differentiates this assumedly abnormal interpolation from what is so assuredly "normal interpolation" as referred to here.